### DISCUSSION TITLE
Running the app.

### DISCUSSION THREAD

**Discussion by souvik-blade** [deckerst/aves#851] [Q&A]
I am unable to run the app on my PC as you have described about adding a reference to the keystore for the credentials which I am not so sure about. Please help!!!

**Comment by deckerst** [comment]
Just to be sure we're on the same page, you're a developer trying to build and run the app from an IDE, right? If so, please provide details, logs, etc.

If you're not a developer but a regular user, please note that the app is for Android only. That said, it's possible to run it on Windows via the Amazon App Store and the Windows Subsystem for Android.

**Reply by souvik-blade** [reply]
Thanks for getting back to me. So regarding the problem, this is what i'm getting in VS code ;
![Screenshot 2023-12-22 180931](https://github.com/deckerst/aves/assets/104029276/19a857be-87df-4e7b-ab8d-afdff3876eb9)

**Reply by deckerst** [reply]
From the logs it's an issue with the `media_kit` dependency on your system.
To move forward i suggest you try the following:
1) delete the `pub` cache at `C:\Users\Souvi\AppData\Local\Pub\Cache\`,
2) clean the flutter project,
3) rebuild.

If it still fails, try to build the test app from `media_kit` and troubleshoot with them your building issues.

**Reply by souvik-blade** [reply]
actually I think we need to provide these fields with specific values, which I am unaware of.
please guide me.
storeFile=<KEYSTORE_PATH>
storePassword=<KEYSTORE_PASSWORD>
keyAlias=<KEY_ALIAS>
keyPassword=<KEY_PASSWORD>
googleApiKey=<GOOGLE_API_KEY>

**Reply by deckerst** [reply]
Correct, but you're not there yet. Fix the issue that appears in your logs first.

**Comment by souvik-blade** [comment]
Now i'm stuck with this after making all the changes that needed to be done.
![ss](https://github.com/deckerst/aves/assets/104029276/c04d6b0d-0273-4018-be64-27b4dbcb60e0)

**Reply by deckerst** [reply]
This dependency is only required for the `huawei` flavor of Aves and will be automatically excluded for the other flavors.

Did you:
1) run the script to apply your flavor of choice? E.g. `scripts/apply_flavor_play.sh` for the `play` flavor
2) run the app with the appropriate command? E.g. `./flutterw run -t lib/main_play.dart --flavor play`

If so, what is the command that yields the error above?

**Reply by souvik-blade** [reply]
This is what I'm getting after using the command that you mentioned though I had to edit the command  a bit as it was not working.
![image](https://github.com/deckerst/aves/assets/104029276/75224faa-570d-45bc-ae37-6d18e80711cf)

**Reply by deckerst** [reply]
You forgot the flavor parameter `--flavor play` in your command.

Also, using your system Flutter instead of the embedded one is not an issue as long as your system Flutter matches the expected version written in pubspec.yaml.

**Reply by souvik-blade** [reply]
And now I'm stuck at this.
![image](https://github.com/deckerst/aves/assets/104029276/990a01d8-4d22-421d-8ab7-766c90837aa1)

**Reply by deckerst** [reply]
Haha, it's interesting to see all the possible build issues one can have. For this one, i guess you need to install a JDK and let gradle know where it is. I don't use VS so I'm not sure how it's configured.

Also, it seems like a basic Android project build issue. I suggest you go through a basic Android hello world tutorial and build that, before you go forward with Aves.
